osu.Game/Screens/Play/Player.cs

```diff
@@ -564,17 +582,23 @@ public void Restart()
         private ScheduledDelegate completionProgressDelegate;
         private Task<ScoreInfo> prepareScoreForDisplayTask;
 
-        private void updateCompletionState(ValueChangedEvent<bool> completionState)
+        /// <summary>
+        /// Handles changes in player state which may progress the completion of gameplay / this screen's lifetime.
+        /// </summary>
+        /// <param name="skipStoryboardOutro">If in a state where a storyboard outro is to be played, offers the choice of skipping beyond it.</param>
+        /// <exception cref="InvalidOperationException">Thrown if this method is called more than once without changing state.</exception>
+        private void updateCompletionState(bool skipStoryboardOutro = false)
         {
             // screen may be in the exiting transition phase.
             if (!this.IsCurrentScreen())
                 return;
 
-            if (!completionState.NewValue)
+            if (!ScoreProcessor.HasCompleted.Value)
             {
                 completionProgressDelegate?.Cancel();
                 completionProgressDelegate = null;
                 ValidForResume = true;
+                skipOutroOverlay.Hide();
```

Comment:
This was all I did to disable the overlay upon rewinding, but after reaching the outro again, the overlay fades out immediately unlike when it is first shown. I think this is caused by the logic in `SkipOverlay.cs` that keeps track of the progress since `displayTime` isn't updated after the overlay is first loaded.

https://github.com/ppy/osu/blob/68807a90ec4f0251252ba728d8d12d1be5201655/osu.Game/Screens/Play/SkipOverlay.cs#L115-L120

Reply:
I'm confused. You're saying it still doesn't work correctly, but the test you added passes. Do you have another test which fails?

Reply:
I've adjusted the test and fixed the underlying issue (state not being passed on to `FadeContainer`, which is the one handling the "fade on user movement" part).